java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -41,6 +41,15 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
       case LOGICAL_COMPLEMENT:
         result = new SymbolicValue.NotSymbolicValue(counter++);
         break;
+      case AND:
+        result = new SymbolicValue.AndSymbolicValue(counter++);
+        break;
+      case OR:
+        result = new SymbolicValue.OrSymbolicValue(counter++);
+        break;
+      case XOR:
+        result = new SymbolicValue.XorSymbolicValue(counter++);
+        break;
       case INSTANCE_OF:
         result = new SymbolicValue.InstanceOfSymbolicValue(counter++);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)